extra_data/reader.py

```diff
@@ -592,18 +592,41 @@ def union(self, *others):
         """
         files = set(self.files)
         train_ids = set(self.train_ids)
+        is_single_run = self.is_single_run
+        run_num = None
+        proposal_num = None
+
+        # Account for old runs (file format 0.5),
+        # which doesn't have these information.
+        if (
+            "runNumber" in self.run_metadata().keys() and
+            "proposalNumber" in self.run_metadata().keys()
+        ):
+            if self.is_single_run:
+                run_num = self.run_metadata()["runNumber"]
+                proposal_num = self.run_metadata()["proposalNumber"]
```

Comment:
We should check that it's a single run before getting the metadata, because `run_metadata` will fail if it's not (and we don't need the metadata then anyway). We can check it for all the pieces we're unioning together at once: `all(dc.is_single_run for dc in (self,) + others)`.

`self.run_metadata()` isn't cached - it calls HDF5 to get the values every time. So it's probably worth calling it once and storing the dict in a local variable.